replica/table.cc

```diff
@@ -2935,7 +2935,13 @@ future<row_locker::lock_holder> table::do_push_view_replica_updates(shared_ptr<d
     const bool need_static = db::view::needs_static_row(m.partition(), views);
     if (!need_regular && !need_static) {
         tracing::trace(tr_state, "View updates do not require read-before-write");
-        co_await generate_and_propagate_view_updates(gen, base, sem.make_tracking_only_permit(s, "push-view-updates-1", timeout, tr_state), std::move(views), std::move(m), { }, tr_state, now);
+        if (_config.materialized_views_rmw_admission()) {
```

Comment:
Maybe we can abstract this in a `table::obtain_mv_rmw_permit()`.

Reply:
I agree. I see in this patch 3 or 4 copies of code that "obtains permits" for MV writes, and as I commented above I don't even remember why all these places exist, but even if they do, maybe they should all call the same function stead of duplicating the logic of get-this-permit-or-that-permit-or-none-at-all.